Sources/MongoSwift/ConnectionString.swift

```diff
@@ -188,4 +192,26 @@ internal class ConnectionString {
 
         return hosts
     }
+
+    internal func setCredential(from credential: Credential) throws {
+        guard mongoc_uri_set_username(self._uri, credential.username) else {
```

Comment:
we probably only want to be calling these libmongoc methods if the value we are passing in is non nil.

Reply:
ah I see you pushed a fix for this right as I published my comments 🙂